_posts/2022-07-22-accessing-host-devices-in-flotta.markdown

```diff
@@ -0,0 +1,322 @@
+
+## Running the workload ##
+
+At this point, we are ready to deploy the workload on the device. For this example, I have created a VM running fedora and attached the webcam in my laptop on `/dev/video2`. I have labeled my edge device with `app=webcam` to make sure that the Flotta controller schedules the workload on this specific device.
+
+```shell
+[jgil@fedora ~]$ kubectl create -f workload_webcam.yaml
+```
+
+Now let's make sure that the workload is running by inspecting the status in the edgedevice:
+
+
+
+```shell
+[jgil@fedora ~]$ kubectl get edgedevice/4cc27d11334f4242baa7efc795d2bbc0 -ojsonpath="{.status.workloads}"| jq .
+[
+  {
+    "lastTransitionTime": "2022-07-14T14:50:28Z",
+    "name": "webcam",
+    "phase": "Running"
+  }
+]
+```
+
+Since I'm running this on a VM, I'm going to ssh to the device and become the `flotta` user to access the container and see that the images were created:
+
+```shell
+[jgil@fedora ~] ssh -l root 192.168.122.23
+[root@fedora ~]# su -l flotta -s /bin/bash -c "podman exec -it webcam-fedora ls -lart /tmp"
+total 112
+dr-xr-xr-x. 1 root root   54 Jul 22 15:02 ..
+-rw-r--r--. 1 root root 8078 Jul 22 15:02 helloworld-1.jpg
+-rw-r--r--. 1 root root 8078 Jul 22 15:02 helloworld-2.jpg
+-rw-r--r--. 1 root root 8068 Jul 22 15:02 helloworld-3.jpg
+-rw-r--r--. 1 root root 9172 Jul 22 15:03 helloworld-4.jpg
+-rw-r--r--. 1 root root 8786 Jul 22 15:03 helloworld-5.jpg
+-rw-r--r--. 1 root root 8699 Jul 22 15:03 helloworld-6.jpg
+-rw-r--r--. 1 root root 8892 Jul 22 15:03 helloworld-7.jpg
+-rw-r--r--. 1 root root 8981 Jul 22 15:03 helloworld-8.jpg
+-rw-r--r--. 1 root root 9358 Jul 22 15:04 helloworld-9.jpg
+-rw-r--r--. 1 root root 9414 Jul 22 15:04 helloworld-10.jpg
+drwxrwxrwt. 1 root root 4096 Jul 22 15:04 .
+```
+
+There we have the 10 images inside the container. As an exercise, you can enhance this example by adding a data sync path to upload the images to a remote S3 storage using the data transfer capabilities of the Flotta agent. This way you won't need to sneak into the VM like I did to see that the images were created.
+
+## Conclusion ##
+
+Flotta is gradually improving the support of workloads that can run on devices without compromising the security of the device. With Flotta it is possible to run workloads that generate and consume data and are capable of synchronizing with a remote compatible S3 storage, as well as workloads that require access to the host mounted devices, such as webcams or sensors.
+
+There is still room for improvement in the usability with the ability to configure the supplementary groups without having to remote to the device each time. We hope in the next releases we can provide a mechanism to configure the host in these areas in a declarative manner. Feel free to drop suggestions or enhancements to the [project](https://github.com/project-flotta/) in the github repository!.
```

Comment:
nit: maybe specific link to issues: https://github.com/project-flotta/flotta-operator/issues

Reply:
good idea. thanks